micro-rdk/src/esp32/provisioning/server.rs

```diff
@@ -0,0 +1,204 @@
+use bytes::Bytes;
+use esp_idf_svc::wifi::{ClientConfiguration, Configuration};
+
+use crate::{
+    common::{
+        app_client::{AppClientBuilder, AppClientConfig},
+        conn::mdns::Mdns,
+        grpc::ServerError,
+        grpc_client::GrpcClient,
+        provisioning::{
+            server::{
+                ProvisioningInfo, ProvisioningService, ProvisioningServiceBuilder,
+                ProvisoningServer, WifiManager,
+            },
+            storage::{RobotCredentialStorage, WifiCredentialStorage},
+        },
+    },
+    esp32::{
+        conn::mdns::Esp32Mdns,
+        exec::Esp32Executor,
+        provisioning::wifi_provisioning::{esp32_get_wifi, Esp32WifiProvisioning},
+        tcp::Esp32Stream,
+        tls::{Esp32TLS, Esp32TLSServerConfig},
+    },
+};
+
+use async_io::{Async, Timer};
+use std::{
+    ffi::CString,
+    fmt::Debug,
+    net::{Ipv4Addr, TcpListener, UdpSocket},
+};
+
+async fn dns_server() {
```

Comment:
not really happy to pull an entire crate just to do DNS so I will sync with clint & ale to check if we can do some mDNS approach